web-portal/pages/login.vue

```diff
@@ -1,10 +1,101 @@
+<template>
+  <div class="container login-page">
+    <h1>Login</h1>
+
+    <div v-if="errorMessage !== null" class="login-page__errors">
+      {{ errorMessage }}
+    </div>
+
+    <form v-on:submit.prevent="onLoginClick">
+      <div class="login-page__field">
+        <label class="login-page__username-label">username: </label>
+        <input
+          v-model="username"
+          type="text"
+          placeholder="username"
+        />
+      </div>
+
+      <div class="login-page__field">
+        <label class="login-page__password-label">password: </label>
+        <input
+          v-model="password"
+          type="password"
+          placeholder="password"
+        />
+      </div>
+
+      <div class="login-page__actions">
+        <input
+          type="submit"
+          class="login-page__login-btn"
+          value="Login"
+        >
+      </div>
+    </form>
+  </div>
+</template>
+
 <script>
+  import getToken from '../store/utils/getToken'
+
   export default {
-    created() {
-      this.$auth.loginWith('auth0')
+    name: 'LoginPage',
+    components: {},
+    data: function () {
+      return {
+        username: '',
+        password: '',
+        errorMessage: null
+      }
     },
-    render() {
-      return null
+    methods: {
+      onLoginClick: function () {
+        const creds = {
+          username: this.username,
+          password: this.password
+        }
+
+        this.$auth.loginWith('local', {
+          data: creds
+        }).then(() => {
+          const token = getToken(this.$auth)
+
+          this.$store.dispatch(
+            'LoadAllUserData',
+            { idToken: token })
+        }).catch((err) => {
+          console.error('error logging in:' + err)
+          this.errorMessage = err
+        })
+      }
     }
   }
 </script>
+
+<style>
```

Comment:
We should probably scope this `<style scoped>`. It doesn't hurt but for consistency.